pkg/api/scylla/v1alpha1/types_scylladbdatacenter.go

```diff
@@ -0,0 +1,528 @@
+
+	// scyllaDB defined ScyllaDB properties for this rack.
+	// These override the settings set on Datacenter level.
+	// +optional
+	ScyllaDB *ScyllaDBTemplate `json:"scyllaDB"`
+
+	// scyllaDBManagerAgent specifies ScyllaDB Manager Agent properties for this rack.
+	// These override the settings set on Datacenter level.
+	// +optional
+	ScyllaDBManagerAgent *ScyllaDBManagerAgentTemplate `json:"scyllaDBManagerAgent"`
+}
+
+// ScyllaDBTemplate allows overriding a subset of ScyllaDB settings.
+type ScyllaDBTemplate struct {
+	// resources requirements for the ScyllaDB container
+	// +optional
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// storage requirements for the containers
+	// +optional
+	Storage *StorageOptions `json:"storage,omitempty"`
+
+	// customConfigMapRef points to custom ScyllaDB configuration stored as ConfigMap.
+	// Overrides upper level settings.
+	// +optional
+	CustomConfigMapRef *string `json:"customConfigMapRef,omitempty"`
+
+	// volumes added to Scylla Pod.
+	// +optional
+	Volumes []corev1.Volume `json:"volumes,omitempty"`
+
+	// volumeMounts to be added to ScyllaDB container.
+	// +optional
+	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
+}
+
+// ScyllaDBManagerAgentTemplate allows to override a subset of ScyllaDBManagerAgent settings.
+type ScyllaDBManagerAgentTemplate struct {
+	// resources are requirements for the ScyllaDB Manager Agent container.
+	// +optional
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// customConfigSecretRef points to custom ScyllaDB Manager Agent configuration stored as Secret.
+	// +optional
+	CustomConfigSecretRef *string `json:"customConfigSecretRef,omitempty"`
+
+	// volumes added to Scylla Pod.
+	// +optional
+	Volumes []corev1.Volume `json:"volumes,omitempty"`
+
+	// volumeMounts to be added to Agent container.
+	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
+}
+
+// RackSpec is the desired state for a ScyllaDB Rack.
+type RackSpec struct {
+	RackTemplate `json:",inline"`
+
+	// name is the name of the ScyllaDB Rack. Used as rack name in GossipingPropertyFileSnitch.
+	// This field is immutable.
+	Name string `json:"name"`
+}
+
+// ScyllaDB holds configuration options related to ScyllaDB.
+type ScyllaDB struct {
+	ScyllaDBTemplate `json:",inline"`
+
+	// image holds a reference to the ScyllaDB container image.
+	Image string `json:"image"`
+
+	// externalSeeds specifies the external seeds to propagate to ScyllaDB binary on startup as "seeds" parameter of seed-provider.
+	ExternalSeeds []string `json:"externalSeeds,omitempty"`
+
+	// alternatorOptions designates this cluster an Alternator cluster.
+	// +optional
+	AlternatorOptions *AlternatorOptions `json:"alternatorOptions,omitempty"`
+
+	// additionalScyllaDBArguments will be appended to the ScyllaDB binary during startup.
+	// When set, ScyllaDB may behave unexpectedly, and every such setup is considered unsupported.
```

Comment:
It's useful, for example changing log level wouldn't make entire cluster unsupported, it's just a message to discourage users from using it if they don't know consequences.

Reply:
I agree with @zimnx - the purpose of the field is to get some extra flexibility, maybe for simple things or things that are not implemented yet. The "unsupported" means you can do crazy things, conflict with the operator but we won't solve your issue if it doesn't work.

Reply:
Ok either way, but maybe the comment could be extended with this explanation (i.e. saying that it might conflict with the operator etc.) and saying something along the lines of consider using customConfigMapRef instead, if possible?

Reply:
added mention about customConfigMapRef

Reply:
all good now, thx